dts/arm/raspberrypi/rpi_pico/rp2350.dtsi

```diff
@@ -0,0 +1,349 @@
+		};
+
+		clk_gpout1: clk-gpout1 {
+			compatible = "raspberrypi,pico-clock";
+			clocks = <&pll_sys>;
+			clock-names = "pll_sys";
+			clock-frequency = <150000000>;
+			#clock-cells = <0>;
+		};
+
+		clk_gpout2: clk-gpout2 {
+			compatible = "raspberrypi,pico-clock";
+			clocks = <&pll_sys>;
+			clock-names = "pll_sys";
+			clock-frequency = <150000000>;
+			#clock-cells = <0>;
+		};
+
+		clk_gpout3: clk-gpout3 {
+			compatible = "raspberrypi,pico-clock";
+			clocks = <&pll_sys>;
+			clock-names = "pll_sys";
+			clock-frequency = <150000000>;
+			#clock-cells = <0>;
+		};
+
+		clk_ref: clk-ref {
+			compatible = "raspberrypi,pico-clock";
+			clocks = <&xosc>;
+			clock-names = "xosc";
+			clock-frequency = <12000000>;
+			#clock-cells = <0>;
+		};
+
+		clk_sys: clk-sys {
+			compatible = "raspberrypi,pico-clock";
+			clocks = <&pll_sys>;
+			clock-names = "pll_sys";
+			clock-frequency = <150000000>;
+			#clock-cells = <0>;
+		};
+
+		clk_usb: clk-usb {
+			compatible = "raspberrypi,pico-clock";
+			clocks = <&pll_usb>;
+			clock-names = "pll_usb";
+			clock-frequency = <48000000>;
+			#clock-cells = <0>;
+		};
+
+		clk_adc: clk-adc {
+			compatible = "raspberrypi,pico-clock";
+			clocks = <&pll_usb>;
+			clock-names = "pll_usb";
+			clock-frequency = <48000000>;
+			#clock-cells = <0>;
+		};
+
+		clk_peri: clk-peri {
+			compatible = "raspberrypi,pico-clock";
+			clocks = <&clk_sys>;
+			clock-names = "clk_sys";
+			clock-frequency = <150000000>;
+			#clock-cells = <0>;
+		};
+
+		pll_sys: pll-sys {
+			compatible = "raspberrypi,pico-pll";
+			clocks = <&xosc>;
+			clock-names = "xosc";
+			clock-div= <1>;
+			fb-div= <125>;
+			post-div1 = <5>;
+			post-div2 = <2>;
+			#clock-cells = <0>;
+		};
+
+		pll_usb: pll-usb {
+			compatible = "raspberrypi,pico-pll";
+			clocks = <&xosc>;
+			clock-names = "xosc";
+			clock-div= <1>;
+			fb-div = <100>;
+			post-div1 = <5>;
+			post-div2 = <5>;
+			#clock-cells = <0>;
+		};
+
+		rosc: rosc {
+			compatible = "raspberrypi,pico-rosc";
+			clock-frequency = <6500000>;
+			range = <RPI_PICO_ROSC_RANGE_RESET>;
+			stage-drive-strength = <0>, <0>, <0>, <0>, <0>, <0>, <0>, <0>;
+			clock-div = <16>;
+			phase = <0>;
+			#clock-cells = <0>;
+		};
+
+		rosc_ph: rosc-ph {
+			compatible = "raspberrypi,pico-clock";
+			clock-frequency = <6500000>;
+			clocks = <&rosc>;
+			clock-names = "rosc";
+			#clock-cells = <0>;
+		};
+
+		xosc: xosc {
+			compatible = "raspberrypi,pico-clock";
+			clock-frequency = <12000000>;
+			#clock-cells = <0>;
+		};
+
+		gpin0: gpin0 {
+			compatible = "raspberrypi,pico-clock";
+			status = "disabled";
+			clock-frequency = <0>;
+			#clock-cells = <0>;
+		};
+
+		gpin1: gpin1 {
+			compatible = "raspberrypi,pico-clock";
+			status = "disabled";
+			clock-frequency = <0>;
+			#clock-cells = <0>;
+		};
+	};
+
+	soc {
+		compatible = "raspberrypi,rp2350", "simple-bus";
+
+		sram0: memory@20000000 {
+			compatible = "mmio-sram";
+			reg = <0x20000000 DT_SIZE_K(264)>;
```

Comment:
This would only allow us to use 264kByte (as we had in the RP2040). The RP2350 has larger SRAM (see https://datasheets.raspberrypi.com/rp2350/rp2350-datasheet.pdf#%5B%7B%22num%22%3A34%2C%22gen%22%3A0%7D%2C%7B%22name%22%3A%22XYZ%22%7D%2C115%2C590.784%2Cnull%5D).
Related: https://github.com/zephyrproject-rtos/zephyr/pull/77368/files#diff-b42a568a700c609cd0b9157a2204aff4f4e903513fbe9f403dfe3d853de357c6R6

Reply:
Nice catch. Fixed.